homeassistant/components/foscam/switch.py

```diff
@@ -0,0 +1,83 @@
+"""This component provides basic support for Foscam IP cameras."""
+from __future__ import annotations
+
+from typing import Any
+
+from libpyfoscam import FoscamCamera
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import FoscamCoordinator
+from .const import DOMAIN, LOGGER
+from .entity import FoscamCoordinatorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up foscam switch from a config entry."""
+
+    session: FoscamCamera = hass.data[DOMAIN][config_entry.entry_id]["session"]
+    coordinator: FoscamCoordinator = hass.data[DOMAIN][config_entry.entry_id][
+        "coordinator"
+    ]
+
+    await coordinator.async_config_entry_first_refresh()
+
+    async_add_entities([FoscamAwakeSwitch(session, coordinator, config_entry)])
+
+
+class FoscamAwakeSwitch(FoscamCoordinatorEntity, SwitchEntity):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(
+        self,
+        session: FoscamCamera,
+        coordinator: FoscamCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize a Foscam camera."""
+        super().__init__(session, coordinator, config_entry)
+
+        self._attr_name = "Awake"
+        self._attr_unique_id = config_entry.entry_id + "_awake_switch"
+
+        self.is_asleep = False
```

Comment:
Shouldn't be this set to `coordinator.data["is_awake"]`? At this point, this should be available and represent the real state, if not there is a delay when HA starts until the first update.